### DISCUSSION TITLE
about finding local package?

### DISCUSSION THREAD

**Discussion by ZiJiaW** [xmake-io/xmake#2147] [Q&A] [answered]
这一页文档 https://xmake.io/#/package/system_package 上的说明：

>Remote download can be forcibly disabled by {system = false}. At this time, it is equivalent to find_package of Xmake/CMake, but it is simpler and easier to use, and it is exactly the same as the use of remote packages.

显示如果设置system=false，则xmake会尝试本地查找包。

我已经在linux上自行安装了liblua5.3，因此我期望`add_requires("liblua5.3", {system = false})`能够帮我链接到lua库，然而它会显示not found，而当我改成`add_requires("liblua5.3", {system = true})`时，却能够正确链接。

实测因为xrepo中lua库的名字不叫liblua，就叫lua，因此如果改成`add_requires("lua", {system = false})`，xmake就会远程下载lua包。

说明system=true时xmake才是disable remote download，system=false时应该是强制remote download，所以这页文档是不是写反了。。。看起来有点confusing

**Comment by waruqi** [comment, marked as answer]
> 我已经在linux上自行安装了liblua5.3，因此我期望add_requires("liblua5.3", {system = false})能够帮我链接到lua库，然而它会显示not found，而当我改成add_requires("liblua5.3", {system = true})时，却能够正确链接。

system = true 才是强制查本地系统库，不走远程安装

> 实测因为xrepo中lua库的名字不叫liblua，就叫lua，因此如果改成add_requires("lua", {system = false})，xmake就会远程下载lua包。

system = false 你显式禁用查找本地系统库，那当然只能走远程安装了。。`add_requires("lua")` 直接这样就行了，默认找系统，找不到走远程，如果装了还是找不到，确认 pkg-config 是否安装 

> 说明system=true时xmake才是disable remote download，system=false时应该是强制remote download，所以这页文档是不是写反了。。。看起来有点confusing

就是这个意思，文档那个写错了，你可以提个 pr 改下
